libs/ledger-live-common/src/exchange/swap/hooks/v5/useFetchCurrencyAll.ts

```diff
@@ -0,0 +1,18 @@
+import { getAvailableProviders } from "../..";
+import { useAPI } from "../../../../hooks/useAPI";
+import { fetchCurrencyAll } from "../../api/v5";
+
+export function useFetchCurrencyAll() {
+  const { data, ...rest } = useAPI({
+    queryFn: fetchCurrencyAll,
+    queryProps: {
+      providers: getAvailableProviders(),
+    },
+    // assume the all currency list for the given props won't change during a users session.
+    staleTimeout: Infinity,
```

Comment:
How long does it take? (only curiosity) ;)

Reply:
Wasn't more than a few ms to a second during development.